src/app.rs

```diff
@@ -37,19 +39,78 @@ pub struct Application<'a, 'b> {
 
 impl<'a, 'b> Application<'a, 'b> {
     /// Creates a new Application with the given initial game state.
-    pub fn new<S: State + 'a>(initial_state: S) -> Result<Application<'a, 'b>> {
+    /// This will create and allocate all the needed resources for
+    /// the event loop of the game engine. It is a shortcut for convenience
+    /// if you need more control over how the engine is configured you should
+    /// be using [build](struct.Application.html#method.build) instead.
+    ///
+    /// # Parameters
+    /// - `initial_state`: The initial State handler of your game See
+    ///   [State](trait.State.html) for more information on what this is.
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` type wrapping the `Application` type. See
+    /// [errors](struct.Application.html#errors) for a full list of
+    /// possible errors that can happen in the creation of a Application object.
+    ///
+    /// # Type Parameters
+    ///
+    /// - `S`: A type that implements the `State` trait. e.g. Your initial
+    ///        game logic. 
+    ///
+    /// # Lifetimes
+    ///
+    /// - `a`: The lifetime of the `State` objects.
+    /// - `b`: This lifetime is inherited from `specs` and `shred`, it is
+    ///        the minimum lifetime of the systems used by `Application`
+    ///
+    /// # Errors
+    ///
+    /// Application will return an error if the internal threadpool fails
+    /// to initialize correctly because of systems resource limitations
+    ///
+    /// # Examples
+    ///
+    /// ~~~no_run
+    /// use amethyst::prelude::*;
+    ///
+    /// struct NullState;
+    /// 
+    /// impl State for NullState {}
+    /// 
+    /// fn main() {
+    ///     let mut game = Application::new(NullState).expect("Failed to initialize");
+    ///     game.run();
+    /// }
+    /// ~~~
+    pub fn new<S>(initial_state: S) -> Result<Application<'a, 'b>>
+    where
+        S: State + 'a
+    {
         ApplicationBuilder::new(initial_state)?.build()
     }
 
-    /// Builds a new Application with the given settings.
+
+    /// Creates a new ApplicationBuilder with the given initial game state.
+    ///
+    /// This is identical in function to [ApplicationBuilder::new](struct.ApplicationBuilder.html#method.new).
     pub fn build<S>(initial_state: S) -> Result<ApplicationBuilder<'a, 'b, S>>
     where
         S: State + 'a,
     {
         ApplicationBuilder::new(initial_state)
     }
 
-    /// Starts the application and manages the game loop.
+    /// Run the gameloop until the game's state indicates that the game is no
```

Comment:
s/gameloop/game loop
s/game's/games

Reply:
> s/game's/games

Are you sure about this?  The game does after all own the state in this sentence does it not?

Reply:
It's a pedantic question, but ultimately the `Application` is what's managing the game state. The game loop doesn't manage the game's state, so technically the original doc comment was more correct.